src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -42,12 +40,42 @@ class GitVersionPlugin implements Plugin<Project> {
             }
         }
 
+        project.ext.versionDetails = {
+            File gitDir = getRootGitDir(project.rootDir)
+
+            try {
+                Git git = Git.wrap(new FileRepository(gitDir))
+                String description = git.describe().call();
+
+                // Description has no git hash so it is just the tag name
+                if(!(description =~ /.*g.?[0-9a-fA-F]{3,}/)) {
+                  return new VersionDetails(description, 0);
+                }
+
+                String[] split = description.split('-')
```

Comment:
I think it'd be cleaner here to use a regular expression then to split things up and sow them back together.